### PR TITLE
converge env/example and env/vagrant

### PR DESCRIPTION
**Pull request by paulczar**
The two key elements of this change is to converge our env/example and env/vagrant paths and have them both use the same systems.   what was in env/example is no env/example/ci and the vagrant stacks are moved from env/vagrant/XXX to env/vagrant/XXX.

I have also moved configuration of vagrant boxes out of the Vagrantfile and into yml files that are declared in each env.    

finally to allow for any differences in required variables I created a /envs/example/vagrant.yml which is overlayed on top of defaults.yml when bin/run_vagrant is called.

### CODE REVIEW

File: test/run [hunk 1]
```diff
@@ -16,10 +16,7 @@ source $(dirname $0)/common
 
 cd ${ROOT}
 
-if [ "${ACTION}" == "all" -o "${ACTION}" == "unit" ]; then
-  tox
-  ring_bell
-fi
+echo cd ${ROOT}
```
Comment (j2sol) [inline test/run]: Is this just going to echo here?

**Comment by j2sol** [comment]
There sure are a lot of various vagrant*.yml files in a bunch of directories :)

This needs a rebase before merge. The ansible bits look fine, I can't really speak for the vagrant bits.

**Comment by masteinhauser** [comment]
Tests failed due to below. Guessing this needs a rebase off of master now?

```
TASK: [nova-data | enable nested kvm] *****************************************
Thursday 16 April 2015  08:42:33 -0700 (0:00:00.232)       0:35:40.525 ********
changed: [173.247.105.23]
changed: [173.247.105.6]
changed: [173.247.105.3]

TASK: [nova-data | install nova-compute packages] *****************************
Thursday 16 April 2015  08:42:34 -0700 (0:00:00.785)       0:35:41.310 ********
failed: [173.247.105.23] => (item=libvirt-bin=1.2.2-0ubuntu13.1.6~cloud2,python-libvirt=1.2.2-0ubuntu2~cloud0,qemu-kvm=2.0.0+dfsg-2ubuntu1.6~cloud0,librbd1=0.80.5-0ubuntu0.14.04.1~cloud0,open-iscsi,libvirt-dev=1.2.2-0ubuntu13.1.6~cloud2,pkg-config,genisoimage) => {"failed": true, "item": "libvirt-bin=1.2.2-0ubuntu13.1.6~cloud2,python-libvirt=1.2.2-0ubuntu2~cloud0,qemu-kvm=2.0.0+dfsg-2ubuntu1.6~cloud0,librbd1=0.80.5-0ubuntu0.14.04.1~cloud0,open-iscsi,libvirt-dev=1.2.2-0ubuntu13.1.6~cloud2,pkg-config,genisoimage"}
stderr: E: Version '1.2.2-0ubuntu13.1.6~cloud2' for 'libvirt-bin' was not found
E: Version '2.0.0+dfsg-2ubuntu1.6~cloud0' for 'qemu-kvm' was not found
E: Version '0.80.5-0ubuntu0.14.04.1~cloud0' for 'librbd1' was not found
E: Version '1.2.2-0ubuntu13.1.6~cloud2' for 'libvirt-dev' was not found

stdout: Reading package lists...
Building dependency tree...
Reading state information...

msg: 'apt-get install 'libvirt-bin=1.2.2-0ubuntu13.1.6~cloud2' 'qemu-kvm=2.0.0+dfsg-2ubuntu1.6~cloud0' 'librbd1=0.80.5-0ubuntu0.14.04.1~cloud0' 'open-iscsi' 'libvirt-dev=1.2.2-0ubuntu13.1.6~cloud2' 'pkg-config' 'genisoimage' ' failed: E: Version '1.2.2-0ubuntu13.1.6~cloud2' for 'libvirt-bin' was not found
E: Version '2.0.0+dfsg-2ubuntu1.6~cloud0' for 'qemu-kvm' was not found
E: Version '0.80.5-0ubuntu0.14.04.1~cloud0' for 'librbd1' was not found
E: Version '1.2.2-0ubuntu13.1.6~cloud2' for 'libvirt-dev' was not found

failed: [173.247.105.3] => (item=libvirt-bin=1.2.2-0ubuntu13.1.6~cloud2,python-libvirt=1.2.2-0ubuntu2~cloud0,qemu-kvm=2.0.0+dfsg-2ubuntu1.6~cloud0,librbd1=0.80.5-0ubuntu0.14.04.1~cloud0,open-iscsi,libvirt-dev=1.2.2-0ubuntu13.1.6~cloud2,pkg-config,genisoimage) => {"failed": true, "item": "libvirt-bin=1.2.2-0ubuntu13.1.6~cloud2,python-libvirt=1.2.2-0ubuntu2~cloud0,qemu-kvm=2.0.0+dfsg-2ubuntu1.6~cloud0,librbd1=0.80.5-0ubuntu0.14.04.1~cloud0,open-iscsi,libvirt-dev=1.2.2-0ubuntu13.1.6~cloud2,pkg-config,genisoimage"}
stderr: E: Version '1.2.2-0ubuntu13.1.6~cloud2' for 'libvirt-bin' was not found
E: Version '2.0.0+dfsg-2ubuntu1.6~cloud0' for 'qemu-kvm' was not found
E: Version '0.80.5-0ubuntu0.14.04.1~cloud0' for 'librbd1' was not found
E: Version '1.2.2-0ubuntu13.1.6~cloud2' for 'libvirt-dev' was not found

stdout: Reading package lists...
Building dependency tree...
Reading state information...

msg: 'apt-get install 'libvirt-bin=1.2.2-0ubuntu13.1.6~cloud2' 'qemu-kvm=2.0.0+dfsg-2ubuntu1.6~cloud0' 'librbd1=0.80.5-0ubuntu0.14.04.1~cloud0' 'open-iscsi' 'libvirt-dev=1.2.2-0ubuntu13.1.6~cloud2' 'pkg-config' 'genisoimage' ' failed: E: Version '1.2.2-0ubuntu13.1.6~cloud2' for 'libvirt-bin' was not found
E: Version '2.0.0+dfsg-2ubuntu1.6~cloud0' for 'qemu-kvm' was not found
E: Version '0.80.5-0ubuntu0.14.04.1~cloud0' for 'librbd1' was not found
E: Version '1.2.2-0ubuntu13.1.6~cloud2' for 'libvirt-dev' was not found

failed: [173.247.105.6] => (item=libvirt-bin=1.2.2-0ubuntu13.1.6~cloud2,python-libvirt=1.2.2-0ubuntu2~cloud0,qemu-kvm=2.0.0+dfsg-2ubuntu1.6~cloud0,librbd1=0.80.5-0ubuntu0.14.04.1~cloud0,open-iscsi,libvirt-dev=1.2.2-0ubuntu13.1.6~cloud2,pkg-config,genisoimage) => {"failed": true, "item": "libvirt-bin=1.2.2-0ubuntu13.1.6~cloud2,python-libvirt=1.2.2-0ubuntu2~cloud0,qemu-kvm=2.0.0+dfsg-2ubuntu1.6~cloud0,librbd1=0.80.5-0ubuntu0.14.04.1~cloud0,open-iscsi,libvirt-dev=1.2.2-0ubuntu13.1.6~cloud2,pkg-config,genisoimage"}
stderr: E: Version '1.2.2-0ubuntu13.1.6~cloud2' for 'libvirt-bin' was not found
E: Version '2.0.0+dfsg-2ubuntu1.6~cloud0' for 'qemu-kvm' was not found
E: Version '0.80.5-0ubuntu0.14.04.1~cloud0' for 'librbd1' was not found
E: Version '1.2.2-0ubuntu13.1.6~cloud2' for 'libvirt-dev' was not found

stdout: Reading package lists...
Building dependency tree...
Reading state information...

msg: 'apt-get install 'libvirt-bin=1.2.2-0ubuntu13.1.6~cloud2' 'qemu-kvm=2.0.0+dfsg-2ubuntu1.6~cloud0' 'librbd1=0.80.5-0ubuntu0.14.04.1~cloud0' 'open-iscsi' 'libvirt-dev=1.2.2-0ubuntu13.1.6~cloud2' 'pkg-config' 'genisoimage' ' failed: E: Version '1.2.2-0ubuntu13.1.6~cloud2' for 'libvirt-bin' was not found
E: Version '2.0.0+dfsg-2ubuntu1.6~cloud0' for 'qemu-kvm' was not found
E: Version '0.80.5-0ubuntu0.14.04.1~cloud0' for 'librbd1' was not found
E: Version '1.2.2-0ubuntu13.1.6~cloud2' for 'libvirt-dev' was not found
```

**Comment by paulczar** [comment]
this may not have actually passed tests ... got run by the new jenkins server I'm to test via PR.

**Pull request merged**